### ISSUE TITLE
While training GRPO, I noticed that my model crashes. Its loss is 0, its grad_norm and kl are both Nan, and it completes as “!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!””

### ISSUE THREAD

**Issue by KevinClaint** [modelscope/ms-swift#3930]
https://private-user-images.githubusercontent.com/100783202/435057451-0408650c-0ab9-4d27-b3f0-503b2ad37c01.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ5NjQ1NTQsIm5iZiI6MTc0NDk2NDI1NCwicGF0aCI6Ii8xMDA3ODMyMDIvNDM1MDU3NDUxLTA0MDg2NTBjLTBhYjktNGQyNy1iM2YwLTUwM2IyYWQzN2MwMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQxOFQwODE3MzRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wYzQwYzY0ZTFiZTIxZmM3OWFjNDc2NjNmMWQ1OTZjNzA3NWQ3OGM2YTFkOWQ0M2YzNmM3Njg5MDE1YzJjOGYwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.4a8bQeJm35oXg5mtDHI4sFY0Fp2jlCn1c_qQpZS2EXY

**Comment by KevinClaint** [comment]
![Image](https://github.com/user-attachments/assets/94ea3d2c-afde-4f54-b10f-6dad17c8eaef)

**Comment by effortprogrammer** [comment]
Did you solve the issue?

**Comment by KevinClaint** [comment]
I turned down my learning rate and then changed the initial REWARD from 0 to 1e-4 (a very small value), after which it didn't happen. Tried and true from other issues as well.
https://github.com/volcengine/verl/issues/747

**Comment by effortprogrammer** [comment]
I don't understand how did you change initial reward from 0 to 1e-4.. Can you give more context with this?

**Comment by KevinClaint** [comment]
> I don't understand how did you change initial reward from 0 to 1e-4.. Can you give more context with this?
> 
> In addition, can you provide which versions of libraries you are currently using?

For example， if the model doesn't output the answer that can get the reward, the reward will be computed as zero.  I just change it into 1e-4

**Comment by hjh0119** [comment]
what's the version of swift? I believe the issue with NaN gradients has been fixed.

 https://github.com/modelscope/ms-swift/blob/main/swift/trainers/mixin.py#L264-L281

**Comment by KevinClaint** [comment]
> My swfit version is 3.3.0 dev0, same problem, the grad clip seems not work. I use deepspeed zero2.
> 
> ![Image](https://github.com/user-attachments/assets/f608b9c1-c00e-40b3-af16-8875ecdc5323)

I also use deepspeed zero2. And I find that my problem can't be solved by using the method mentioned above. It must be something wrong, but it comes casually. I can't be stable to re-produce this problem again.

**Comment by KevinClaint** [comment]
I also find that it occurs in the first 2k steps (most of them are in 1k steps). And after 2k steps, I never meet it.

**Comment by zhangansen** [comment]
我把我的temperature设为0，一开始就出现!!!!!!

**Comment by zhangansen** [comment]
温度设为其他就没事了

**Comment by hjh0119** [comment]
Feel free to reopen if you have any issues.
